doc/rst/technotes/gpu.rst

```diff
@@ -123,6 +123,9 @@ The following are further requirements for GPU support:
 * For targeting NVIDIA or AMD GPUs, ``LLVM`` must be used as Chapel's backend
   compiler (i.e.  ``CHPL_LLVM`` must be set to ``system`` or ``bundled``).
 
+  * Note that ``CHPL_TARGET_COMPILER`` must be ``llvm``. This is the default
+    when ``CHPL_LLVM`` is set to ``system`` or ``bundled``.
```

Comment:
Don't we also need to say that `CHPL_TARGET_CC` needs to actually point at the LLVM compiler?

This was the mistake the Spack package was making, essentially setting `CHPL_TARGET_COMPILER=llvm CHPL_TARGET_CC=gcc CHPL_TARGET_CXX=g++`.

Maybe this isn't the best place to say this, but I think somewhere needs to say "those three variables, if set, all need to match", especially since the scripting doesn't currently seem to enforce that invariant..

Reply:
I am not sure this is the best place to put that information. I think the best place for that would be [here](https://chapel-lang.org/docs/main/usingchapel/chplenv.html#cc-and-similar).

As far as having the chplenv scripts just enforce that, I think that pretty reasonable and have opened [XXX](https://github.com/chapel-lang/chapel/pull/25300) to add it (post-release at this point)